ansible/roles-infra/infra-osp-resources-destroy/tasks/heat.yml

```diff
@@ -1,16 +1,16 @@
 ---
 - environment:
     OS_AUTH_URL: "{{ osp_auth_url }}"
-    OS_USERNAME: "{{ osp_auth_username }}"
-    OS_PASSWORD: "{{ osp_auth_password }}"
+    OS_USERNAME: "{{ osp_auth_username_member | d(osp_auth_username) }}"
+    OS_PASSWORD: "{{ osp_auth_password_member | d(osp_auth_password) }}"
     OS_PROJECT_NAME: "{{ osp_project_name }}"
     OS_PROJECT_DOMAIN_ID: "{{ osp_auth_project_domain }}"
     OS_USER_DOMAIN_NAME: "{{ osp_auth_user_domain }}"
   block:
   # There is no module that allows retrieving this data.
   - name: Get a list of stacks in the project
     command: >-
-      openstack stack list --property tenant={{ osp_project_info[0].id }} -f value --column ID
+      openstack stack list --property tenant={{ osp_project_info[0].id | d(osp_project_id) }} -f value --column ID
```

Comment:
I think these should be reversed. Use `osp_project_id` if defined...otherwise fall back to `default(osp_project_info[0].id`. We'll only get `osp_project_info` if `osp_project_id` is not defined anyhow.

Also, I think this could be set as a fact in a task before to make it a little easier to read. WDYT?

Reply:
That was my initial plan :) It should be good now.